ignite/metrics/precision.py

```diff
@@ -250,31 +322,32 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
             indices = torch.argmax(y_pred, dim=1).view(-1)
             y_pred = to_onehot(indices, num_classes=num_classes)
         elif self._type == "multilabel":
-            # if y, y_pred shape is (N, C, ...) -> (C, N x ...)
-            num_classes = y_pred.size(1)
-            y_pred = torch.transpose(y_pred, 1, 0).reshape(num_classes, -1)
-            y = torch.transpose(y, 1, 0).reshape(num_classes, -1)
+
+            num_labels = y_pred.size(1)
+            y_pred = torch.transpose(y_pred, 1, -1).reshape(-1, num_labels)
+            y = torch.transpose(y, 1, -1).reshape(-1, num_labels)
```

Comment:
Why it is better to make it (N * ..., C) instead of (C, N * ...) ?

Reply:
Because in binary and multilabel case, y and y_pred finally become of size N*C. I did the same thing to multilabel to reuse the logic in following lines.